### PR TITLE
fix(types): Replace custom TYPE_CHECKING with stdlib typing.TYPE_CHECKING

### LINKED ISSUE #3426: Get rid of `sentry_sdk._types.TYPE_CHECKING`
We define our own `TYPE_CHECKING` constant here: https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/_types.py This seems to be due to backwards compatibility with older Python versions.

The stdlib `typing` module added [`TYPE_CHECKING`](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING) in 3.5.2 (we support 3.6+), we can get rid of our alias and use `typing.TYPE_CHECKING` directly.

### PR DESCRIPTION
**Pull request by dev-satoshi**
This commit replaces the custom TYPE_CHECKING constant in _types.py with the standard library's typing.TYPE_CHECKING. This change simplifies the code and removes unnecessary backward compatibility for Python versions below 3.6. Since the _types module is private, this update does not introduce any breaking changes.

Closes #3426

### CODE REVIEW

**Comment by dev-satoshi** [comment]
@sentrivana 
Got it! I’ll take care of it!

File: sentry_sdk/client.py
Comment (dev-satoshi) [inline sentry_sdk/client.py:1]: There are two import statements in this file, so it's better to remove one of them, right?

File: sentry_sdk/_types.py [hunk 1]
```diff
@@ -1,7 +1,4 @@
-try:
-    from typing import TYPE_CHECKING
-except ImportError:
-    TYPE_CHECKING = False
```
Comment (dev-satoshi) [inline sentry_sdk/_types.py:4]: I removed the try-except block since we are now supporting Python 3.6 and above.

File: sentry_sdk/client.py
Comment (dev-satoshi) [inline sentry_sdk/client.py:1]: There are two import statements in this file, so it's better to remove one of them, right?

**Review by dev-satoshi — COMMENTED** [review]
@sentrivana
It's done!
Could you please review this when you have a moment?

File: sentry_sdk/client.py
Comment (dev-satoshi) [inline sentry_sdk/client.py:1]: There are two import statements in this file, so it's better to remove one of them, right?

File: sentry_sdk/_types.py [hunk 2]
```diff
@@ -1,7 +1,4 @@
-try:
```
Comment (dev-satoshi) [inline sentry_sdk/_types.py:1]: I removed the try-except block since we are now supporting Python 3.6 and above.

File: sentry_sdk/_types.py [hunk 3]
```diff
@@ -1,7 +1,4 @@
...
-    from typing import TYPE_CHECKING
-except ImportError:
-    TYPE_CHECKING = False
+from typing import TYPE_CHECKING
```
Comment (dev-satoshi) [inline sentry_sdk/_types.py:1]: I removed the try-except block since we are now supporting Python 3.6 and above.

**Review by sentrivana — APPROVED** [review]
Thank you @dev-satoshi! Looks good to me :)

**Comment by sentrivana** [comment]
Just fyi, I've pushed a [commit](https://github.com/getsentry/sentry-python/pull/3447/commits/8fd52deb5264122cfd023567e281704f340a6483) that reorders some of the imports (e.g. if the `typing` import is in the middle of `sentry_sdk` imports, I move it directly to the `if TYPE_CHECKING` block). Will merge this as soon as the tests are happy.

**Comment by sentrivana** [comment]
All is looking good except for the AWS Lambda tests (this has nothing to do with this PR, we need to fix this on master). We'll probably get around to fixing this early next week and then we can merge this.

Nothing additional needed from your side @dev-satoshi -- thanks again for the contribution, we'll see that this is merged next week!

**Pull request merged**